e2e/etcd_test_with_cov.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e_cov
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/pkg/expect"
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/coreos/etcd/pkg/flags"
+)
+
+func spawnCmd(args []string) (*expect.ExpectProcess, error) {
+	if args[0] == binPath {
+		var (
+			prefix  = "ETCD"
+			cmd     = binDir + "/etcd_test"
+			envArgs []string
+		)
+		for i := range args[1:] {
+			if strings.Contains(args[i], "--") {
+				flag := strings.Split(args[i], "--")[1]
+				val := "true"
+				// split the flag that has "="
+				// e.g --auto-tls=true" => flag=auto-tls and val=true
+				if strings.Contains(args[i], "=") {
+					split := strings.Split(flag, "=")
+					flag = split[0]
+					val = split[1]
+				}
+
+				if i+1 < len(args) {
+					if !strings.Contains(args[i+1], "--") {
+						val = args[i+1]
+					}
+				}
+				envArgs = append(envArgs, fmt.Sprintf("%s=%s", flags.FlagToEnv(prefix, flag), val))
```

Comment:
No need to sprintf, string concatenation is fine:
`append(envArgs, flags.FlagToEnv(prefix, flag) + "=" + val)`